config.go

```diff
@@ -52,8 +52,8 @@ var logger = logrus.Logger{
 //		+ debugMode bool: true => run client in debug mode.
 //		+ sync bool: true => client is in synchronous mode. The client will block until Create/Update/Delete processes.
 //		are completely finished. It is safer to set this parameter to `true`.
-//		+ delayIntervalMilliSecs int: delay (in milliseconds) between requests when checking request (or retry 5xx, 424 error code).
-//		+ maxNumberOfRetries int: number of retries when server returns 5xx, 424 error code.
+//		+ delayIntervalMilliSecs int: delay (in milliseconds) between requests when checking request (or retry 503 error code).
+//		+ maxNumberOfRetries int: number of retries when server returns 503 error code.
```

Comment:
From what I understand from the code this documentation is not correct. Also when I look at `retryNTimes` invocation, the code is different from what is spec-ed in #200.

So my questions:
- In case of 429 is `maxNumberOfRetries` taken into account? I believe it should not, and it is not documented here but `retryNTimes` function does not really differentiate between 429 and 503.

Goals are as follows as per #200 :
- On 429, retry _indefinitely_ until the timestamp given in `RateLimit-Reset` has been reached
- On 503, retry a maximum of `maxNumberOfRetries` which should be something like 5 or 10 by default. If `Retry-After: <delay-seconds>` is given, retry after that timestamp has been reached. If `Retry-After` is not given or is `Retry-After: <http-date>` (unexpectedly) use `delayIntervalMilliSecs`

Reply:
@bkircher. Answer: Requests with 429 (rate-limit is reached) response code will also be retried when the rate-limit is reset. If the rate-limit is reached again, retry again in when the rate-limit is reset. There is no `maxNumberOfRetries` for 429 request retry. Check L266-267.